File: selenium-java-e2e-template/src/main/java/pages/LoginPage.java
package pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;

public class LoginPage extends BasePage {
    private static final By USERNAME_INPUT = By.cssSelector("input[name='username'], input#username, input[type='email']");
    private static final By PASSWORD_INPUT = By.cssSelector("input[name='password'], input#password");
    private static final By SUBMIT_BUTTON = By.cssSelector("button[type='submit'], input[type='submit']");
    
    public LoginPage(WebDriver driver) {
        super(driver);
    }
    
    public void login(String username, String password) {
        waitForElement(USERNAME_INPUT).sendKeys(username);
        driver.findElement(PASSWORD_INPUT).sendKeys(password);
        driver.findElement(SUBMIT_BUTTON).click();
    }
}

File: web/selenium-python/pages/login_page.py
from selenium.webdriver.common.by import By
from .base_page import BasePage

class LoginPage(BasePage):
    USERNAME_INPUT = (By.CSS_SELECTOR, 'input[name="username"], input#username, input[type="email"]')
    PASSWORD_INPUT = (By.CSS_SELECTOR, 'input[name="password"], input#password')
    SUBMIT_BUTTON = (By.CSS_SELECTOR, 'button[type="submit"], input[type="submit"]')
    
    def login(self, username, password):
        self.wait_for_element(self.USERNAME_INPUT).send_keys(username)
        self.driver.find_element(*self.PASSWORD_INPUT).send_keys(password)
        self.driver.find_element(*self.SUBMIT_BUTTON).click()